pipelines/matomo_pipeline.py

```diff
@@ -0,0 +1,71 @@
+"""Contains functions that run the matomo pipeline."""
+from matomo import matomo_reports, matomo_events
+import dlt
+from time import time
+
+
+def run_full_load() -> None:
+    """
+    Does a basic run of the pipeline.
+    """
+    pipeline_reports = dlt.pipeline(dataset_name="matomo_full_load", full_refresh=False, destination="postgres", pipeline_name="matomo")
+    data_reports = matomo_reports()
+    data_events = matomo_events()
+    info = pipeline_reports.run([data_reports, data_events])
+    print(info)
+
+
+def run_custom_reports():
+    """
+    Defines some custom reports you can use and shows how to use for different custom reports
+    :return:
+    """
+
+    queries = [
+        {"resource_name": "custom_report_name",
+         "methods": ["CustomReports.getCustomReport"],
+         "date": "2020-01-01",
+         "period": "day",
+         "site_id": 2,
+         "extra_params": {"idCustomReport": 1}},
+        {"resource_name": "custom_report_name2",
+         "methods": ["CustomReports.getCustomReport"],
+         "date": "2020-01-01",
+         "period": "day",
+         "site_id": 2,
+         "extra_params": {"idCustomReport": 2}},
+    ]
+    pipeline_reports = dlt.pipeline(dataset_name="matomo_custom_reports", full_refresh=False, destination="postgres", pipeline_name="matomo")
+    data = matomo_reports(queries=queries)
+    info = pipeline_reports.run(data)
+    print(info)
+
+
+def run_reports():
+    """
+    Runs the pipeline only loading reports.
+    :return:
+    """
+    pipeline_events = dlt.pipeline(dataset_name="matomo_reports", full_refresh=False, destination="postgres", pipeline_name="matomo")
+    data = matomo_reports()
```

Comment:
i'd expect most of the people will pass site id explicitly. maybe you could do that here, not relying on configuration

Reply:
@rudolfix, For reports site_id is inside the queries. I guess you also want to move site_id out of the queries and have the same site_id for every query?